### DISCUSSION TITLE
Material ui TextInput display error with fetch data

### DISCUSSION THREAD

**Discussion by Kublick** [react-hook-form/react-hook-form#2326] [Q&A] [answered]
Helllo

I stumbled into an issue,  using material ui textfiled for my inputs ; when I try to make an update form and fetching the users info into the the inputs the label is sitting along the input values ; its like the fetch info doesnt trigger an onchange on the textfield to move the label to the expectedl position.

Example here : https://codesandbox.io/s/naughty-bhabha-yekjn?file=/src/App.js&fbclid=IwAR3G-Dk8SJCUHpbCfiNGDNa9rivCPmBNT-PD3X5Bra4NapgARgwLFEfqVDo

Just tap on fetch 

p.d. Lib is awesome !!

**Comment by bluebill1049** [comment, marked as answer]
thanks mate, just use controller: https://react-hook-form.com/api#Controller will resolve this problem.

**Reply by Kublick** [reply]
Hi Bill ; I tried with control before posting here and got same behavior sometimes it does work but most of the time it doesnt.. updated the example with my implementation

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/agitated-lumiere-le8vv?file=/src/App.js

**Reply by Kublick** [reply]
Thanks ; I see now what I had incorrectly on my side

On my project I was saving the data to an state so the form can load it thus generating the error ; now switching to Controller and passing the data directly from the table seems to do the trick 👍 

cheers
